rust/main/chains/hyperlane-cosmos/src/providers/cosmos/provider.rs

```diff
@@ -178,12 +241,17 @@ impl CosmosProvider {
             .fold("".to_string(), |acc, denom| acc + ", " + denom);
 
         if !unsupported_denominations.is_empty() {
-            error!(
+            let msg = "transaction contains fees in unsupported denominations, manual intervention is required";
+            warn!(
                 ?tx_hash,
                 ?supported_denomination,
                 ?unsupported_denominations,
-                "transaction contains fees in unsupported denominations, manual intervention is required");
+                msg,
+            );
+            Err(ChainCommunicationError::CustomError(msg.to_owned()))?
```

Comment:
I'm now remembering more details of what I poorly documented in https://github.com/hyperlane-xyz/issues/issues/1369, flagging just in case we're not on the same page

Maybe this is your understanding as well but the issue is that if we propagate an Err here, it propagates very far back and prevents the indexed data (e.g. the messages indexed in a range), and any other data in that block range batch, from being written to the DB. If you trace back the call sites all the way to `store_logs` it'll be more clear

So i think this is fine as long as we accept that all data in batches where a tx has this problem won't get written, and we plan to revisit this

Reply:
I agree that we can start with such implementation: refrain from storing logs from transaction we cannot parse properly. It will allows us to add the support for missing features and re-visit the transaction without a need for a manual backfill of data.

I created a few issues preliminary: https://github.com/hyperlane-xyz/hyperlane-monorepo/issues/4632, https://github.com/hyperlane-xyz/hyperlane-monorepo/issues/4633, https://github.com/hyperlane-xyz/hyperlane-monorepo/issues/4634